### PR TITLE
Translated for portuguese

### PR DESCRIPTION
**Pull request by tavaresgerson**

### CODE REVIEW

File: pt/development/testing.rst [hunk 1]
```diff
@@ -70,32 +70,1775 @@ usada pelo CakePHP para fixar tabelas e dados::
...
+
+        $this->get('/api/posts');
+        $this->assertResponseOk();
+    }
+
+A chave de cabeçalhos em ``configRequest()`` pode ser usada para configurar 
+qualquer cabeçalho HTTP adicional necessário para uma ação.
+
+Testando Ações Protegidas por CsrfComponent ou SecurityComponent
+---------------------------------------------------------------
```
Comment (othercorey) [inline pt/development/testing.rst]:
```suggestion
----------------------------------------------------------------
```

File: pt/orm/query-builder.rst [hunk 2]
```diff
@@ -1,22 +1,1503 @@
...
+criada usando o sistema de convenções::
+
+    // Dentro de ArticlesController.php
+
+    $query = $this->Articles->find();
+
+Selecionando Linhas de uma Tabela
+---------------------------------
+
+::
```
Comment (othercorey) [inline pt/orm/query-builder.rst]:
```suggestion
.. code-block::
```

File: pt/orm/query-builder.rst [hunk 3]
```diff
@@ -1,22 +1,1503 @@
...
+        ->find()
+        ->where(['id' => 1])
+        ->first();
+
+    debug($article->title);
+
+Obtendo uma Lista de Valores de uma Coluna
+------------------------------------------
+
+::
```
Comment (othercorey) [inline pt/orm/query-builder.rst]:
```suggestion
.. code-block::
```

File: pt/development/testing.rst [hunk 4]
```diff
@@ -70,32 +70,1775 @@ usada pelo CakePHP para fixar tabelas e dados::
...
+uma fixture para cada modelo que terá uma consulta executada nele. Para carregar a fixture, 
+defina a propriedade ``$fixtures`` no seu modelo::
+
+    class ArticlesTest extends TestCase
+    {
+        public $fixtures = ['app.Articles', 'app.Comments'];
+    }
+
+O item acima carregará os fixtures de Article e Coment do 
+diretório fixture do aplicativo. Você também pode carregar fixture do core do CakePHP ou plugins:
```
Comment (markstory) [inline pt/development/testing.rst]:
```suggestion
diretório fixture do aplicativo. Você também pode carregar fixture do core do CakePHP ou plugins::
```

File: pt/development/testing.rst [hunk 5]
```diff
@@ -70,32 +70,1775 @@ usada pelo CakePHP para fixar tabelas e dados::
...
+    $this->assertEquals('jose', $user->username);
+
+    // Afirmar cookies na resposta
+    $this->assertCookie('1', 'thingid');
+
+    // Verifique o tipo de conteúdo
+    $this->assertContentType('application/json');
+
+Além dos métodos de asserção acima, você também pode usar todas as asserções no `TestSuite
+<https://api.cakephp.org/3.x/class-Cake.TestSuite.TestCase.html>` e
```
Comment (markstory) [inline pt/development/testing.rst]:
```suggestion
<https://api.cakephp.org/3.x/class-Cake.TestSuite.TestCase.html>`__ e 
```
The trailing `__` are needed to create a link.

File: pt/development/testing.rst [hunk 6]
```diff
@@ -70,32 +70,1775 @@ usada pelo CakePHP para fixar tabelas e dados::
...
+
+    // Afirmar cookies na resposta
+    $this->assertCookie('1', 'thingid');
+
+    // Verifique o tipo de conteúdo
+    $this->assertContentType('application/json');
+
+Além dos métodos de asserção acima, você também pode usar todas as asserções no `TestSuite
+<https://api.cakephp.org/3.x/class-Cake.TestSuite.TestCase.html>` e 
+os encontrados em `PHPUnit <https://phpunit.de/manual/current/en/appendixes. assertions.html>`.
```
Comment (markstory) [inline pt/development/testing.rst]:
```suggestion
os encontrados em `PHPUnit <https://phpunit.de/manual/current/en/appendixes. assertions.html>`__.
```
The trailing `__` are needed to create a link.

File: pt/development/testing.rst [hunk 7]
```diff
@@ -70,32 +70,1775 @@ usada pelo CakePHP para fixar tabelas e dados::
...
+==============
+
+Geralmente a maioria dos aplicativos não testa diretamente seu código HTML. 
+Fazer isso geralmente resulta em conjuntos de testes frágeis e difíceis de 
+manter, com tendência a serem quebrados. Ao escrever testes funcionais usando 
+:php:class:`IntegrationTestTrait`, você pode inspecionar o conteúdo da 
+visualização renderizada configurando a opção` `return`` para 'view'. Embora 
+seja possível testar o conteúdo da visualização usando ``IntegrationTestTrait``, 
+um teste de integração/visualização mais robusto e sustentável pode ser realizado 
+usando ferramentas como `Selenium webdriver <http://seleniumhq.org>`.
```
Comment (markstory) [inline pt/development/testing.rst]:
```suggestion
usando ferramentas como `Selenium webdriver <http://seleniumhq.org>`__.
```
The trailing `__` are needed to create a link.

**Comment by markstory** [comment]
@tavaresgerson Let me know when you'd like this merged.

**Comment by othercorey** [comment]
Nice work, thanks!

**Comment by tavaresgerson** [comment]
> @tavaresgerson Let me know when you'd like this merged.

One moment :-)

File: pt/orm/query-builder.rst [hunk 8]
```diff
@@ -1,22 +1,1502 @@
...
+    }
+
+Para obter mais informações sobre como personalizar os campos usados para preencher a lista, 
+consulte seção :ref:`table-find-list`.
+
+As Consultas são Objetos de Coleção
+-----------------------------------
+
+Depois de se familiarizar com os métodos do objeto Query, é altamente recomendável 
+que você visite a seção :doc:`Coleção </ core-libraries / collections>` para
```
Comment (othercorey) [inline pt/orm/query-builder.rst]:
```suggestion
que você visite a seção :doc:`Coleção </core-libraries/collections>` para 
```

File: pt/orm/query-builder.rst [hunk 9]
```diff
@@ -1,22 +1,1502 @@
...
+------------------------------------------
+
+Objetos de consulta são avaliados preguiçosamente. Isso significa que uma consulta 
+não é executada até que ocorra uma das seguintes coisas:
+
+- A consulta é iterada com ``foreach()``.
+- O método ``execute()`` da consulta é chamado. Isso retornará o objeto subjacente
+   de instrução e deve ser usado com consultas de inserção/atualização/exclusão.
+- O método ``first()`` da consulta é chamado. Isso retornará o primeiro resultado no conjunto
+   construído por ``SELECT`` (ele adiciona `` LIMIT 1`` à consulta).
```
Comment (othercorey) [inline pt/orm/query-builder.rst]:
```suggestion
   construído por ``SELECT`` (ele adiciona ``LIMIT 1`` à consulta).
```

File: pt/orm/query-builder.rst [hunk 10]
```diff
@@ -1,22 +1,1502 @@
...
+
+    // Seleciona todos os campos da tabela de artigos, 
+    // incluindo um campo slug calculado.
+    $query = $articlesTable->find();
+    $query
+        ->select(['slug' => $query->func()->concat(['title' => 'identifier', '-', 'id' => 'identifier'])])
+        ->select($articlesTable); // Select all fields from articles
+
+.. versionadded:: 3.1
+    Passar um objeto de tabela para select () foi adicionado em 3.1.
```
Comment (othercorey) [inline pt/orm/query-builder.rst]:
```suggestion
    Passar um objeto de tabela para select() foi adicionado em 3.1.
```

File: pt/orm/query-builder.rst (hunk 9 above)
Comment (tavaresgerson) [reply pt/orm/query-builder.rst]: Fixed

**Comment by tavaresgerson** [comment]
> @tavaresgerson Let me know when you'd like this merged.

Done! :-)

File: pt/orm/query-builder.rst [hunk 11]
```diff
@@ -1,22 +1,1500 @@
...
+        ])
+        ->values([
+            'title' => 'Second post',
+            'body' => 'Another body text'
+        ])
+        ->execute();
+
+Geralmente, é mais fácil inserir dados usando entidades e
+:php:meth:`~Cake\\ORM\\Table::save()`. Ao compor uma consulta ``SELECT`` e ``INSERT`` juntas, 
+você pode criar consultas de estilo  ``INSERT INTO ... SELECT``
```
Comment (othercorey) [inline pt/orm/query-builder.rst]:
```suggestion
você pode criar consultas de estilo  ``INSERT INTO ... SELECT``::
```

File: pt/orm/query-builder.rst [hunk 12]
```diff
@@ -1,22 +1,1500 @@
...
+
+    $expr = $query->newExpr()->add($userData);
+    $query->select(['two' => $expr]);
+
+Valores de Ligação
+------------------
+
+É possível proteger contra muitas situações inseguras usando ligações. 
+Semelhante a :ref:`vinculando valores a instruções preparadas <database-basics-binding-values>`, 
+os valores podem ser vinculados a consultas usando o método :php:meth:`Cake\\Database\\Query::bind()`
```
Comment (othercorey) [inline pt/orm/query-builder.rst]:
```suggestion
os valores podem ser vinculados a consultas usando o método :php:meth:`Cake\\Database\\Query::bind()`.
```

File: pt/orm/query-builder.rst [hunk 13]
```diff
@@ -1,22 +1,1500 @@
...
+        ->where(['comment LIKE' => '%CakePHP%']);
+
+    $query = $articles->find()
+        ->where(['id IN' => $matchingComment]);
+
+Subqueries are accepted anywhere a query expression can be used. For example, in
+the ``select()`` and ``join()`` methods.
+
+Subconsultas são aceitas em qualquer lugar em que uma expressão de consulta possa 
+ser usada. Por exemplo, nos métodos ``select()`` e ``join()``
```
Comment (othercorey) [inline pt/orm/query-builder.rst]:
```suggestion
ser usada. Por exemplo, nos métodos ``select()`` e ``join()``.
```

File: pt/orm/query-builder.rst [hunk 14]
```diff
@@ -1,22 +1,1500 @@
...
+
+A maioria dos fornecedores de bancos de dados relacionais suporta a remoção 
+de bloqueios ao executar operações selecionadas. Você pode usar o método 
+``epilog()`` para este::
+
+    // Em MySQL
+    $query->epilog('FOR UPDATE');
+
+O método ``epilog()`` permite anexar SQL bruto ao final das consultas. 
+Você nunca deve colocar dados brutos do usuário em ``epilog()``
```
Comment (othercorey) [inline pt/orm/query-builder.rst]:
```suggestion
Você nunca deve colocar dados brutos do usuário em ``epilog()``.
```

**Comment by othercorey** [comment]
Looks good to me!

**Pull request merged**